### PR TITLE
Support for multiple zabbix servers and individual port setting for each server.

### PR DESCRIPTION
**Pull request by erikoest**
Please review. And thanks for your suggestions. Ruby is really not my native tongue, so feel free to rewrite and improve.

### CODE REVIEW

File: lib/puppet/reports/zabbix.rb [hunk 1]
```diff
@@ -11,28 +11,45 @@ def process
...
-    Puppet.debug "sending zabbix report for host #{self.host}, at #{zabbix_sender.serv}:#{zabbix_sender.port}"
-    result = zabbix_sender.send! host_overrides.fetch(self.host, self.host)
+      # send metrics to zabbix
+      Puppet.debug "sending zabbix report for host #{self.host}, at #{zabbix_sender.serv}:#{zabbix_sender.port}"
+      result = zabbix_sender.send! host_overrides.fetch(self.host, self.host)
+
+      # validate the response. if it fails, keep on sending to all
+      # zabbix servers before reporting the error.
+      if result['response'] != 'success'
+        Puppet.error "zabbix send failed - #{result['info']}" if result['response'] != 'success'
```
Comment (shamil) [inline lib/puppet/reports/zabbix.rb]: no need for `if result['response'] != 'success'`, the `if` above already does it

File: zabbix.yaml [hunk 2]
```diff
@@ -1,6 +1,11 @@
 ---
-:zabbix_host: zabbix.domain.tld
-:zabbix_port: 10051
+:zabbix_host:
```
Comment (shamil) [inline zabbix.yaml:2]: This should be `:zabbix_hosts`, keep old `:zabbix_host` unchanged, and just merge it to the `zabbix_hosts` array inside the login

File: lib/puppet/util/zabbix_sender.rb [hunk 3]
```diff
@@ -64,4 +64,4 @@ def rawdata(data)
     end
 
   end
-end
```
Comment (shamil) [inline lib/puppet/util/zabbix_sender.rb:67]: This file should not contain any changes, please reset it

**Review by shamil — CHANGES_REQUESTED** [review]
Generally looks good. Couple of changes and we are ready to merge it. 

I can merge it now, and do the changes myself.

Also, please squash your commits, to have single commit for this PR

**Comment by erikoest** [comment]
On 11/16/2016 05:01 PM, Alex Simenduev wrote:

> _@shamil_ requested changes on this pull request.
> 
> Generally looks good. Couple of changes and we are ready to merge it.
> 
> I can merge it now, and do the changes myself.

Thanks.

> Also, please squash your commits, to have single commit for this PR

I don't think it is recommended to squash commits which have already
been pushed (and possibly shared with others). Wouldn't it be better if
you merge it and squash the commits before you push the change on your
repo?

/ Erik

**Comment by shamil** [comment]
Alright, I've squashed your commits into single one and applied some other changes.
Closing this PR. Thanks for contribution!

**Pull request closed without merging**